Looms.PoS/Looms.PoS.Application/Features/OrderItem/Commands/UpdateOrderItem/UpdateOrderItemCommandHandler.cs

```diff
@@ -0,0 +1,108 @@
+using Looms.PoS.Application.Interfaces;
+using Looms.PoS.Application.Interfaces.ModelsResolvers;
+using Looms.PoS.Application.Models.Requests;
+using Looms.PoS.Domain.Daos;
+using Looms.PoS.Domain.Exceptions;
+using Looms.PoS.Domain.Interfaces;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using System.Transactions;
+
+// TODO: Add rollback if a different product variation is added to update the quantity of the previous product variation
+
+namespace Looms.PoS.Application.Features.OrderItem.Commands.UpdateOrderItem;
+
+public class UpdateOrderItemCommandHandler : IRequestHandler<UpdateOrderItemCommand, IActionResult>
+{
+    private readonly IOrderItemsRepository _orderItemsRepository;
+    private readonly IDiscountsRepository _discountsRepository;
+    private readonly IOrderItemModelsResolver _modelsResolver;
+    private readonly IHttpContentResolver _httpContentResolver;
+    private readonly IProductsRepository _productsRepository;
+    private readonly IProductVariationRepository _variationsRepository;
+    private readonly IProductModelsResolver _productModelsResolver;
+    private readonly IProductVariationModelsResolver _productVariationModelsResolver;
+
+    public UpdateOrderItemCommandHandler(IOrderItemsRepository orderItemsRepository, 
+        IDiscountsRepository discountsRepository,
+        IOrderItemModelsResolver modelsResolver,
+        IHttpContentResolver httpContentResolver,
+        IProductModelsResolver productModelsResolver,
+        IProductVariationModelsResolver productVariationModelsResolver,
+        IProductsRepository productsRepository,
+        IProductVariationRepository variationsRepository)
+    {
+        _orderItemsRepository = orderItemsRepository;
+        _discountsRepository = discountsRepository;
+        _modelsResolver = modelsResolver;
+        _httpContentResolver = httpContentResolver;
+        _productsRepository = productsRepository;
+        _variationsRepository = variationsRepository;
+        _productModelsResolver = productModelsResolver;
+        _productVariationModelsResolver = productVariationModelsResolver;
+    }
+
+    public async Task<IActionResult> Handle(UpdateOrderItemCommand command, CancellationToken cancellationToken)
+    {
+        var orderItemRequest = await _httpContentResolver.GetPayloadAsync<UpdateOrderItemRequest>(command.Request);
+        var originalDao = await _orderItemsRepository.GetAsync(Guid.Parse(command.Id));
+        var productVariation = orderItemRequest.VariationId is null ? null : await _variationsRepository.GetAsync(Guid.Parse(orderItemRequest.VariationId));
+        var quantityToDeduct = orderItemRequest.Quantity - originalDao.Quantity;
+
+        if (quantityToDeduct != 0 && originalDao.Product.Quantity < quantityToDeduct)
```

Comment:
this should be done in validator

Reply:
all 3 checks should be done in validator

Reply:
added to validator